flax/testing/benchmark.py

```diff
@@ -0,0 +1,239 @@
+# Lint as: python3
+
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark class for Flax regression and integration testing.
+
+This file defines utility functions for collecting model training results
+from TensorBoard summaries, and for reporting benchmarks in a JSON format
+for pickup by continuous integration / monitoring frameworks.
+
+When the `benchmark_output_dir` is provided, the benchmark results are
+saved in this directory in the JSON format with a single file per
+benchmark.
+"""
+
+import functools
+import itertools
+import inspect
+import json
+from typing import Dict
+import os
+from absl import flags
+from absl import logging
+from absl.testing import absltest
+
+from tensorboard.backend.event_processing import directory_watcher
+from tensorboard.backend.event_processing import event_file_loader
+
+
+flags.DEFINE_string(
+  'benchmark_output_dir', default=None, help='Benchmark output directory.')
+
+
+FLAGS = flags.FLAGS
+
+
+def _make_events_generator(path):
+  """Makes a generator yielding TensorBoard events from files in `path`."""
+  return directory_watcher.DirectoryWatcher(
+      path,
+      event_file_loader.EventFileLoader).Load()
+
+
+def _process_event(event):
+  """Parse TensorBoard scalars into a (tag, wall_time, step, scalar) tuple."""
```

Comment:
Curious if description should also be included here?

Reply:
What do you mean that each event tuple should include a description?

Reply:
My bad, I think its being done correctly. I thought that the description of the metrics reported by a test case as an 'extra', if we should include it here. I didn't realize it was being considered different from the metrics. 

Took a closer look at benchmark.py, sorry about that. Thanks!